dfc/target.go

```diff
@@ -426,7 +426,7 @@ existslocally:
 	written, err := io.CopyBuffer(w, file, buf)
 	if err != nil {
 		errstr = fmt.Sprintf("Failed to send file %s, err: %v", fqn, err)
-		t.invalmsghdlr(w, r, errstr)
+		t.invalmsghdlr(w, r, errstr, false)
```

Comment:
so this means all places that io.CopyBuffer() is called and failed, no need to write out a http error because it is already failed?
nice finding

Reply:
The issue is actually that a new status code _can't_ be written to the writer because that header has already been written by `io.CopyBuffer` (since it calls `Write()`, which calls `WriteHeader()`). If `WriteHeader()` is called again, it returns a long and messy error.